rust-runtime/aws-smithy-observability/src/provider.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Definitions of high level Telemetry Providers.
+
+use std::sync::Arc;
+
+use crate::{meter::MeterProvider, noop::NoopMeterProvider};
+
+/// A struct to hold the various types of telemetry providers.
+#[non_exhaustive]
+pub struct TelemetryProvider {
+    meter_provider: Box<dyn MeterProvider + Send + Sync>,
+}
+
+impl TelemetryProvider {
+    /// Returns a builder struct for [TelemetryProvider]
+    pub fn builder() -> TelemetryProviderBuilder {
+        TelemetryProviderBuilder {
+            meter_provider: Box::new(NoopMeterProvider),
+        }
+    }
+
+    /// Get the set [MeterProvider]
+    pub fn meter_provider(&self) -> &(dyn MeterProvider + Send + Sync) {
```

Comment:
should this return `impl MeterProvider + Send + Sync`?

Reply:
I don't think you can return a reference to an `impl Trait`, and I didn't want to hand ownership of the `MeterProvider` to the caller, so settled for the `&dyn` that is there now.